runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -1003,6 +1079,13 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
                cg()->supportsNonHelper(TR::SymbolReferenceTable::atomicSwapSymbol);
          case TR::java_lang_Class_isAssignableFrom:
             return cg()->supportsInliningOfIsAssignableFrom();
+         case TR::java_lang_Class_cast:
+            // Do not transform in involuntary OSR, since the resulting
+            // checkcast will still be an OSR point, but it won't be the call
+            // that is expected to correspond to the bytecode instruction. (It
+            // might turn out that there's a safe way to transform anyway, but
+            // it isn't obvious.)
+            return comp()->getOSRMode() != TR::involuntaryOSR;
```

Comment:
In addition to this, can you add a more specific means of disabling this particular transformation (say by an environment variable)?